tfhe/src/strings/server_key/mod.rs

```diff
@@ -243,21 +267,27 @@ impl ServerKey {
 
         let result = self.if_then_else_parallelized(
             &shift_ge_than_str,
-            &self.create_trivial_zero_radix(uint.blocks().len()),
+            &self.create_trivial_zero_radix(len),
             &shifted,
         );
 
         FheString::from_uint(result, false)
     }
 
     fn right_shift_chars(&self, str: &FheString, shift: &RadixCiphertext) -> FheString {
-        let uint = str.to_uint(self);
+        let uint = str.to_uint();
         let mut shift_bits = self.scalar_left_shift_parallelized(shift, 3);
 
         // `shift_bits` needs to have the same block len as `uint` for the tfhe-rs shift to work
         self.pad_or_trim_ciphertext(&mut shift_bits, uint.blocks().len());
 
-        let shifted = self.right_shift_parallelized(&uint, &shift_bits);
+        let len = uint.blocks().len();
+
+        let shifted = if len == 0 {
+            uint
+        } else {
+            self.right_shift_parallelized(&uint, &shift_bits)
+        };
```

Comment:
This is a note for later: As we now have block rotation available, it might be better, so this should be measured at some point

Reply:
Openned at https://github.com/zama-ai/tfhe-rs-internal/issues/829

Reply:
thanks